spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/cache/CacheAutoConfigurationTests.java

```diff
@@ -97,6 +99,16 @@
 
 	private AnnotationConfigApplicationContext context;
 
+	@BeforeClass
+	public static void init() {
+		System.setProperty("hazelcast.jcache.provider.type", "server");
+	}
+
+	@AfterClass
+	public static void close() {
+		System.clearProperty("hazelcast.jcache.provider.type");
+	}
+
```

Comment:
Yes, it's due to client being on the classpath and Hazelcast prefers it in that case. See [`HazelcastCachingProvider#createClientProvider`](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/cache/HazelcastCachingProvider.java#L102-L109).